src/formatDate.js

```diff
@@ -7,8 +7,61 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat[toFormat.length - 1];
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+
+  const dateParts = date.split(oldSeparator);
+
+  const yearIndexFrom =
+    fromFormat.indexOf('YYYY') !== -1
+      ? fromFormat.indexOf('YYYY')
+      : fromFormat.indexOf('YY');
+  const yearIndexTo =
+    toFormat.indexOf('YYYY') !== -1
+      ? toFormat.indexOf('YYYY')
+      : toFormat.indexOf('YY');
+  const monthIndexFrom = fromFormat.indexOf('MM');
+  const dayIndexFrom = fromFormat.indexOf('DD');
+
+  let year = dateParts[yearIndexFrom];
+  const month = dateParts[monthIndexFrom];
+  const day = dateParts[dayIndexFrom];
+
+  if (fromFormat[yearIndexFrom] === 'YY' && toFormat[yearIndexTo] === 'YYYY') {
+    if (year < 30) {
```

Comment:
The comparison `year < 30` will not work as expected because `year` is a string and the comparison is made with a number. This will lead to incorrect comparisons due to type coercion. You should convert `year` to a number before making the comparison.